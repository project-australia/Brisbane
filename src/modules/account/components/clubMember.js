import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View, ScrollView, Text, Alert } from 'react-native'
import { Navbar } from '../../shared/components/navbar'
import { FormOutlineButton } from '../../shared/components/buttons'
import { payWithPayPal } from '../../../services/paypal'
import { styles } from './styles/clubMember.style'

export class ClubMember extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    goBack: PropTypes.func.isRequired,
    logOut: PropTypes.func.isRequired,
    updateProfile: PropTypes.func.isRequired,
    buttonText: PropTypes.string.isRequired,
    user: PropTypes.object.isRequired,
    club: PropTypes.oneOf(['TEN', 'TWENTY', 'NONE']).isRequired,
    price: PropTypes.number.isRequired
  }

  static navigationOptions = {
    title: 'Join our club membership',
    header: null
  }

  show10club = () => {
    return (
      <ScrollView style={styles.wrapper}>
        <Text style={styles.text}>• Get 10% More for books you sell</Text>
        <Text style={styles.text}>• Get 10% Off of purchases</Text>
        <Text style={styles.text}>• Get 10% Off of rentals</Text>
      </ScrollView>
    )
  }

  show20club = () => {
    return (
      <ScrollView style={styles.wrapper}>
        <Text style={styles.text}>• Get 20% More for books you sell</Text>
        <Text style={styles.text}>• Get 20% Off of purchases</Text>
        <Text style={styles.text}>• Get 20% Off of rentals</Text>
      </ScrollView>
    )
  }

  showRep = () => {
    return (
      <ScrollView style={styles.wrapper}>
        <Text style={styles.text}>• Just sign people up to the app (friends, classmates, coworkes, teammates, etc.)</Text>
        <Text style={styles.text}>• When they buy, sell or rent you get a commiission</Text>
        <Text style={styles.text}>• If they become a rep too, you even get a commission from the people they sign up</Text>
        <Text style={styles.text}>• The more you sign up the better!</Text>
        <Text style={styles.text}>• It's so easy!</Text>
      </ScrollView>
    )
  }

  render() {
    return (
      <View style={{flex: 1}}>
        <Navbar
          title={this.props.title}
          onBack={this.props.goBack}
          ignoreAndroidStatusBar
        />
        {this.props.club === 'NONE' && this.show10club()}
        {this.props.club === 'TEN' && this.show20club()}
        {this.props.club === 'TWENTY' && this.showRep()}
        {this.props.club !== 'TWENTY' && <FormOutlineButton
          title={this.props.buttonText}
          onPress={this.checkoutWithPaypal}
          style={styles.input}
        />
        }
      </View>
    )
  }
//   <View style={styles.buttonGroup}>
//   <FlatButton
//     secondary
//     containerStyle={styles.button}
//     title={this.props.buttonText}
//     onPress={this.checkoutWithPaypal}
//   />
// </View>
  checkoutWithPaypal = async () => {
    try {
      await payWithPayPal(
        this.props.price,
        this.props.title,
        this.onPayPalOnSuccess
      )
    } catch (error) {
      console.log('Paypal checkout failed', JSON.stringify(error))
      this.alert('Sorry. Request Failed')
    }
  }

  onPayPalOnSuccess = async paypalResponse => {
    const { user, club, updateProfile, navigate } = this.props
    await updateProfile(user.id, { club })
    this.alert('Successfully registered', navigate('Home'))
  }

  alert = (msg, press = () => {}) =>
    Alert.alert(
      this.props.title,
      msg,
      [
        {
          text: 'Ok',
          onPress: () => press
        }
      ],
      { cancelable: false }
    )
}
