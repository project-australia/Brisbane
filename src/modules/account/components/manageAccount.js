import React, { Component } from 'react'
import { View } from 'react-native'

export class ManageAccount extends Component {
  render () {
    return <View />
  }
}
