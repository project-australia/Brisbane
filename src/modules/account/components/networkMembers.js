import React from 'react'
import { ScrollView, Text, View } from 'react-native'

import { Navbar } from '../../shared/components/navbar'

import { styles } from './styles/networkMembers.style'

const renderMember = member => (
  <View key={member.id} style={styles.wrap}>
    <Text numberOfLines={1} style={styles.title}>
      {member.name}
    </Text>
    <Text numberOfLines={1} style={styles.subtitle}>
      {member.school}
    </Text>
    <Text numberOfLines={1} style={styles.subtitle}>
      {member.email}
    </Text>
  </View>
)

export const NetworkMembers = ({ navigateBack, network }) => {
  return (
    <View style={styles.container}>
      <Navbar title={'Network Members'} onBack={navigateBack} />
      <ScrollView>{network.map(renderMember)}</ScrollView>
    </View>
  )
}
