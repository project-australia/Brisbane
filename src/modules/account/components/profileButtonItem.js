import React from 'react'
import { View } from 'react-native'

import { MenuTitle } from '../../shared/components/menuTitle'
import { FlatButton } from '../../shared/components/buttons'

import { styles } from './styles/profileItem.style'

export const ProfileButtonItem = ({ title, buttonTitle, onPress, secondary }) => {
  return (
    <View style={styles.wrap}>
      <MenuTitle title={title} />
      <FlatButton
        secondary={secondary}
        title={buttonTitle}
        containerStyle={styles.itemBackground}
        onPress={onPress}
      />
    </View>
  )
}
