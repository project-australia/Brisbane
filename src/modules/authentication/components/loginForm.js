import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View } from 'react-native'
import { FormHeader } from './formHeader'
import { styles } from './styles/loginFormStyles'
import { Colors } from '../../../constants'
import { FormTextInput } from './formTextInput'
import { FormButton } from '../../shared/components/buttons'
import { LoadingOverlay } from '../../shared/components/loadingOverlay'

export class LoginForm extends Component {
  static defaultProps = { footer: <View /> }
  static propTypes = {
    footer: PropTypes.object,
    buttonText: PropTypes.string.isRequired,
    alert: PropTypes.shape({
      showAlert: PropTypes.bool.isRequired,
      message: PropTypes.string
    }).isRequired
  }

  // FIXME: Test data only
  state = {
    email: 'eduardomoroni@gmail.com',
    password: '123123',
    loading: false
  }

  setEmail = email => this.setState({ email })
  setPassword = password => this.setState({ password })

  onButtonPress = async () => {
    this.setState({ loading: true })
    const { email, password } = this.state
    await this.props.onButtonPress(email, password)
    this.setState({ loading: false })
  }

  componentWillReceiveProps (nextProps) {
    const { showAlert, message } = nextProps.alert
    if (showAlert) {
      alert(message)
    }
  }

  render () {
    return (
      <LoadingOverlay style={styles.screen} isLoading={this.state.loading}>
        <FormHeader />
        <FormTextInput
          onChangeText={this.setEmail}
          value={this.state.email}
          placeholder="Email address"
          autoCapitalize="none"
          selectionColor={Colors.secondary500}
          style={styles.itemSpacing}
          keyboardType="email-address"
        />
        <FormTextInput
          onChangeText={this.setPassword}
          value={this.state.password}
          placeholder="Password"
          secureTextEntry
          selectionColor={Colors.secondary500}
          style={styles.itemSpacing}
        />
        <FormButton
          title={this.props.buttonText}
          onPress={this.onButtonPress}
          style={styles.itemSpacing}
        />
        {this.props.footer}
      </LoadingOverlay>
    )
  }
}
