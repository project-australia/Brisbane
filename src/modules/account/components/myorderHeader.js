import React from 'react'
import { Text, View } from 'react-native'
import { styles } from './styles/networkMembers.style'

const property = ({ title, property }) => (
  <Text key={title} style={styles.propertyTitle}>
    {title}: <Text style={styles.propertyText}>{property}</Text>
  </Text>
)

export const MyOrderHeader = ({ title, properties }) => (
  <View style={[styles.wrap, styles.standardCardPadding]}>
    <Text style={styles.title}>{title}</Text>
    <View style={styles.textBlockWrap}>{properties.map(property)}</View>
  </View>
)
