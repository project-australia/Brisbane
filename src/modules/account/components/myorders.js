import React from 'react'
import { ScrollView, Text, View } from 'react-native'

import { Navbar } from '../../shared/components/navbar'

import { styles } from './styles/networkMembers.style'

const renderOrder = order => (
  <View key={order.id} style={styles.wrap}>
    <Text numberOfLines={1} style={styles.title}>
      {order.id}
    </Text>
    {order.items.map(book => (
      <Text key={book.id} numberOfLines={1} style={styles.title}>
        {book.title}
      </Text>
    ))}
  </View>
)

export const MyOrders = ({ navigateBack, orders }) => {
  return (
    <View style={styles.container}>
      <Navbar title={'My Orders'} onBack={navigateBack} />
      <ScrollView>{orders.map(renderOrder)}</ScrollView>
    </View>
  )
}
