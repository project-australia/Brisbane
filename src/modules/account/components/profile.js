import React from 'react'
import { ScrollView, View } from 'react-native'
import moment from 'moment'

import { ModalWithInputProfile } from '../../shared/components/modals/modalWithInputProfile'
import { Navbar } from '../../shared/components/navbar'
import { ProfileItem } from './profileItem'
import { ProfileItemAddress } from './profileItemAddress'
import { ScreenTitleAndSubtitle } from './screenTitleAndSubtitle'
import { ProfileRedirectItem } from './profileRedirectItem'
import { ProfileButtonItem } from './profileButtonItem'

import { styles } from './styles/profile.style'

const getSubscriptionText = subType => {
  switch (subType) {
    case 'TEN':
      return '10% off'
    case 'TWENTY':
      return '20% off'
    default:
      return 'Standard'
  }
}

const setUserNetworkNumber = network =>
  network && network.length > 0 && `${network.length} registered members`

export const Profile = ({
  onBackPress,
  user,
  modalTitle,
  navigateToNetwork,
  navigateToWallet,
  onLogoutPress,
  showEditModal,
  isEditModalOpen,
  onConfirmModal,
  onDismissModal
}) => {
  const rightIcon = [
    {
      name: 'pencil',
      onPress: () => showEditModal('Address')
    }
  ]
  return (
    <View style={styles.container}>
      <Navbar onBack={onBackPress} rightIcons={rightIcon} />
      <ScrollView>
        <ScreenTitleAndSubtitle
          title={user.name}
          subtitle={`${getSubscriptionText(user.subscription)} member`}
        />
        <ProfileItem
          title={'School'}
          data={user.school}
          emptyDataLabel={'Add school name'}
          onPress={() => showEditModal('School')}
        />
        <ProfileItem
          title={'Phone'}
          data={user.telephone}
          emptyDataLabel={'Add phone number'}
          onPress={() => showEditModal('Phone')}
        />
        <ProfileItem
          title={'Birth date'}
          data={moment(user.birthDate).format('MMM Do YYYY')}
          emptyDataLabel={'Add birth date'}
          onPress={() => showEditModal('Birth date')}
        />
        <ProfileItemAddress
          title={'Address'}
          data={user.address}
          emptyDataLabel={'Add address'}
        />
        <ProfileRedirectItem
          title={'Network'}
          data={setUserNetworkNumber(user.network)}
          onPress={navigateToNetwork}
        />
        <ProfileButtonItem
          secondary
          title={'My Wallet'}
          buttonTitle={'View my Wallet'}
          onPress={navigateToWallet}
        />
        <ProfileButtonItem buttonTitle={'Logout'} onPress={onLogoutPress} />
      </ScrollView>
      <ModalWithInputProfile
        visible={isEditModalOpen}
        title={modalTitle}
        user={user}
        onConfirm={onConfirmModal}
        onDismiss={onDismissModal}
      />
    </View>
  )
}
