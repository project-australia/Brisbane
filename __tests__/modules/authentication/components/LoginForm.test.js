import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { LoginForm } from '../../../../src/modules/authentication/components/loginForm'

const props = {
  buttonText: 'foo-bar',
  alert: { showAlert: false, message: '' }
}

describe('<LoginForm />', () => {
  xit('should have a snapshot', () => {
    const wrapper = shallow(<LoginForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
