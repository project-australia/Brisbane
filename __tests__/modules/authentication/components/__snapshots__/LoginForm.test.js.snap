// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should have a snapshot 1`] = `
<LoadingOverlay
  isLoading={false}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "HelveticaNeue",
          "fontSize": 26,
        }
      }
    >
      Ballard Books
    </Text>
  </View>
  <FormTextInput
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder="Email Address.js"
    value="eduardomoroni@gmail.com"
  />
  <FormTextInput
    onChangeText={[Function]}
    placeholder="Password"
    secureTextEntry={true}
    value="123456"
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Button
      onPress={[Function]}
      style={
        Object {
          "marginVertical": 10,
        }
      }
      title="foo-bar"
    />
  </View>
  <View />
</LoadingOverlay>
`;
